### PR TITLE
helm: Add AWS database auto-discovery to teleport-kube-agent

### PR DESCRIPTION
**Pull request by webvictim**
This PR implements AWS database auto-discovery in the `teleport-kube-agent` Helm chart.

It also updates `PodSecurityPolicy` for both the `teleport-cluster` and `teleport-kube-agent` Helm charts from `v1beta1` to `v1` to avoid an upcoming deprecation which was causing linter failures.

Closes #10329
Closes #10078
Closes #9857

Thank you to the authors of the above PRs for inspiration and some of the code.

Backports required:
- `branch/v8`

### CODE REVIEW

File: docs/pages/kubernetes-access/helm/reference.mdx [hunk 1]
```diff
@@ -1480,6 +1480,74 @@ You can specify multiple apps by adding additional list elements.
...
+
+  <Admonition type="note">
+    Note that when using `--set` syntax, YAML list elements must be indexed starting at `0`.
+  </Admonition>
+
+  </TabItem>
+</Tabs>
+
+<Admonition type="tip" title="Supported values">
+  You can see a list of all the supported [values which can be used in a Teleport database service configuration here](../../database-access/reference/configuration.mdx).
```
Comment (ptgott) [inline docs/pages/kubernetes-access/helm/reference.mdx:1548]: If the matchers in the `aws` object are the only config sections relevant here, though, I think we can remove this Admonition and explain the `types`, `regions`, and `tags` fields of the matcher object.

**Review by r0mant — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Review by deusxanima — APPROVED** [review]

**Comment by r0mant** [comment]
@webvictim Looks like this PR (in particular, updating PSP version to `policy/v1`) broke teleport-kube-agent installation. @jimbishopp and I ran into this when going through the v9 test plan:

```
 $ helm install teleport-agent ./teleport-kube-agent \
        --set kubeClusterName=mini1-iot \
        --set proxyAddr=host.minikube.internal:3080 \
        --set authToken=hello \
        --set insecureSkipProxyTLSVerify=true \
        --set labels.kubeCluster=mini1-iot \
        --set labels.cluster=root \
        --create-namespace \
        --namespace=teleport
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "PodSecurityPolicy" in version "policy/v1"
```

Changing it back to `policy/v1beta1` fixes the issue. Should we rollback this change?

**Comment by webvictim** [comment]
@r0mant I'll raise a separate PR to revert this; the other changes are too important.

**Comment by r0mant** [comment]
@webvictim Yep, that sounds good. Thank you!

**Comment by webvictim** [comment]
@r0mant

https://github.com/gravitational/teleport/pull/10563
https://github.com/gravitational/teleport/pull/10564
https://github.com/gravitational/teleport/pull/10565

**Pull request merged**